File: examples/log-senders/java/src/main/java/de/sasd/logsink/examples/sender/PostTextLog.java
package de.sasd.logsink.examples.sender;

import java.time.OffsetDateTime;

/**
 * Sendet eine einfache Textmeldung.
 *
 * Die aktuelle V0/V1 speichert den Body unverändert. Deshalb ist text/plain als
 * Testfall weiterhin sinnvoll.
 */
public final class PostTextLog {

    private PostTextLog() {
    }

    public static void main(String[] args) throws Exception {
        LogSinkHttpClient client = LogSinkHttpClient.fromEnvironment();

        String payload = OffsetDateTime.now()
            + " INFO java-text-sender Hello from plain text Java sender";

        System.out.println("POST " + client.baseUrl());
        System.out.println(payload);
        System.out.println();

        HttpResult result = client.post(
            payload,
            "text/plain; charset=utf-8",
            "SASD-java-text-sender/0.1"
        );

        LogSinkHttpClient.printResponse(result);

        if (!result.isSuccessful()) {
            System.exit(1);
        }
    }
}

File: examples/log-senders/php/post-text-log.php
#!/usr/bin/env php
<?php

declare(strict_types=1);

require __DIR__ . '/src/LogSinkClient.php';

$logsinkUrl = getenv('LOGSINK_URL') ?: 'http://api.sasd.de/logsink/index.php';
$client = new LogSinkClient($logsinkUrl);

$payload = sprintf('%s INFO php-text-sender Hello from plain text PHP sender', date(DATE_ATOM));

echo 'POST ' . $logsinkUrl . PHP_EOL;
echo $payload . PHP_EOL . PHP_EOL;

$response = $client->post(
    payload: $payload,
    contentType: 'text/plain; charset=utf-8',
    userAgent: 'SASD-php-text-sender/0.1'
);

LogSinkClient::printResponse($response);

exit(LogSinkClient::isSuccessful($response) ? 0 : 1);
